app/src/main/java/org/p2p/wallet/striga/ui/TopUpWalletPresenter.kt

```diff
@@ -11,29 +9,20 @@ import org.p2p.wallet.common.feature_toggles.toggles.remote.StrigaSignupEnabledF
 import org.p2p.wallet.common.mvp.BasePresenter
 import org.p2p.wallet.infrastructure.network.provider.SeedPhraseProvider
 import org.p2p.wallet.infrastructure.network.provider.SeedPhraseSource
-import org.p2p.wallet.striga.signup.model.StrigaUserStatus
-import org.p2p.wallet.striga.ui.TopUpWalletContract.BankTransferNavigationTarget
-import org.p2p.wallet.striga.user.interactor.StrigaUserInteractor
-import org.p2p.wallet.striga.user.model.StrigaUserVerificationStatus
+import org.p2p.wallet.striga.user.model.StrigaUserStatusDestination
+import org.p2p.wallet.striga.user.repository.StrigaUserStatusRepository
 import org.p2p.wallet.user.interactor.UserInteractor
 
 class TopUpWalletPresenter(
     private val appFeatureFlags: InAppFeatureFlags,
     private val userInteractor: UserInteractor,
     private val strigaSignupFeatureToggle: StrigaSignupEnabledFeatureToggle,
     private val seedPhraseProvider: SeedPhraseProvider,
-    private val strigaUserInteractor: StrigaUserInteractor,
+    private val strigaUserStatusRepository: StrigaUserStatusRepository
 ) : BasePresenter<TopUpWalletContract.View>(),
     TopUpWalletContract.Presenter {
 
     private val strigaBankTransferProgress = MutableSharedFlow<Boolean>(replay = 1)
```

Comment:
Please pay attention when merging with the develop branch to metadata loading. We must ensure that the user cannot proceed until the metadata has fully loaded